### PR TITLE
[0.13 backport] Precompute sighashes

### PR DESCRIPTION
**Pull request by gmaxwell**
applies cleanly

### CODE REVIEW

File: src/main.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ class CScriptCheck;
 class CTxMemPool;
 class CValidationInterface;
 class CValidationState;
-class CachedHashes;
+class PrecomputedTransactionData;
```
Comment (paveljanik) [inline src/main.h]: This will be changed to

```c
struct PrecomputedTransactionData;
```
in the master soon (@sipa).

Comment (btcdrak) [inline src/main.h]: In #8651

**Comment by sipa** [comment]
I believe the arguments to CheckInputs changed, so this is not a clean cherry-pick.

**Comment by gmaxwell** [comment]
will fix.

**Pull request closed without merging**